netsim/augment/links.py

```diff
@@ -1035,13 +1035,21 @@ def interface_feature_check(nodes: Box, defaults: Box) -> None:
     features = devices.get_device_features(ndata,defaults)
     for ifdata in ndata.get('interfaces',[]):
       if 'ipv4' in ifdata:
-        if isinstance(ifdata.ipv4,bool) and ifdata.ipv4 and \
-            not features.initial.ipv4.unnumbered:
-          log.error(
-            text=f'Device {ndata.device} does not support unnumbered IPv4 interfaces',
-            more_hints=[ f'Used on node {node} interface {ifdata.ifname} (link {ifdata.name})' ],
-            category=log.IncorrectValue,
-            module='interfaces')
+        if isinstance(ifdata.ipv4,bool) and ifdata.ipv4:
+          if not features.initial.ipv4.unnumbered:
+            log.error(
+              text=f'Device {ndata.device} does not support unnumbered IPv4 interfaces',
+              more_hints=[ f'Used on node {node} interface {ifdata.ifname} (link {ifdata.name})' ],
+              category=log.IncorrectValue,
+              module='interfaces')
+          elif features.initial.ipv4.unnumbered == 'peer':
+            if len(ifdata.neighbors)!=1 or ifdata.neighbors[0].get('ipv4',False) is not True:
```

Comment:
From the pure packet forwarding perspective, does the remote end have to be an unnumbered link? Admittedly, OSPF wouldn't work.

Reply:
On Linux the remote can be a regular ipv4 prefix, but I decided that the semantics for ```peer``` should be that both sides are unnumbered to simplify things.

Otherwise, the remote would need a route to the unnumbered source loopback address, defeating the purpose of having an unnumbered interface for sake of simplicity